assignments/lambda-classes.js

```diff
@@ -1 +1,221 @@
 // CODE here for your Lambda Classes
+class Person {
+    constructor(personInfo) {
+      this.name = personInfo.name;
+      this.age = personInfo.age;
+      this.location = personInfo.location;
+      this.gender = personInfo.gender;
+      
+    }
+
+    speak() {
+        console.log(`Hello my name is ${this.name}, I am from ${this.location}`)
+    }
+}
+
+class Instructor extends Person {
+    constructor(instructorInfo) {
+      super(instructorInfo);
+      this.specialty = instructorInfo.specialty;
+      this.favLanguage = instructorInfo.favLanguage;
+      this.catchPhrase = instructorInfo.catchPhrase;
+    }
+
+    demo(subject) {
+      console.log(`Today we are learning about ${subject}`);
+    }
+
+    grade(student, subject) {
+      console.log(`${student.name} recieves a perfect score on ${subject}`);
+    }
+  }
+
+class Student extends Person {
+    constructor(studentInfo) {
+        super(studentInfo);
+        this.previousBackground = studentInfo.previousBackground;
+        this.className = studentInfo.previousBackground;
+        this.favSubjects = studentInfo.favSubjects;
+    }
```

Comment:
you may not want to hardcode favorite subjects to console log index 0,1,2, because the array can change sizes, maybe you can do something like: return/ console.log(this.favSubjects)